test/uk/gov/hmrc/economiccrimelevyregistration/controllers/actions/AuthorisedActionSpec.scala

```diff
@@ -36,9 +38,19 @@ class AuthorisedActionSpec extends SpecBase {
     Future(Ok("Test"))
   }
 
+  private val expectedRetrievals =
+    Retrievals.internalId and Retrievals.externalId and Retrievals.confidenceLevel and Retrievals.nino and Retrievals.saUtr and
+      Retrievals.mdtpInformation and Retrievals.credentialStrength and Retrievals.loginTimes and
+      Retrievals.credentials and Retrievals.name and Retrievals.dateOfBirth and Retrievals.email and
+      Retrievals.affinityGroup and Retrievals.agentCode and Retrievals.agentInformation and Retrievals.credentialRole and Retrievals.groupIdentifier and
+      Retrievals.itmpName and Retrievals.itmpDateOfBirth and Retrievals.itmpAddress
+
   "invokeBlock" should {
-    "execute the block and return the result if authorised" in {
-      when(mockAuthConnector.authorise[Option[String]](any(), any())(any(), any())).thenReturn(Future(Some("id")))
+    "execute the block and return the result if authorised" in forAll(
+      arbAuthRetrievals(Some(alphaNumericString)).arbitrary
+    ) { authRetrievals: AuthRetrievals =>
```

Comment:
```suggestion
    ) { authRetrievals =>
```

Reply:
Added type annotation to other test rather than removing this one for the reason we discussed